lsp-diagnostics.el

```diff
@@ -202,10 +209,13 @@ provided by lsp-mode.
 See https://github.com/emacs-lsp/lsp-mode."
       :start #'lsp-diagnostics--flycheck-start
       :modes '(lsp-placeholder-mode) ;; placeholder
-      :predicate (lambda () lsp-mode)
+      :predicate (lambda () lsp-diagnostics-mode)
```

Comment:
this is a bug fix because otherwise the lsp flycheck checker will ill be activated even if it's in lsp-diagnostics-disabled-modes